### ISSUE TITLE
The new created editor plugin need reloading the editor or toggle plugin active status twice to work

### ISSUE THREAD

**Issue by 1Googol** [godotengine/godot#84502]
### Godot version

v4.1.3.stable.official [f06b6836a]

### System information

Godot v4.1.3.stable - Windows 10.0.22000 - Vulkan (Forward+) - dedicated NVIDIA GeForce RTX 3060 (NVIDIA; 30.0.14.7212) - Intel(R) Core(TM) i5-10400F CPU @ 2.90GHz (12 Threads)

### Issue description

*  The new created plugin is not working with or without active.
*  `plugin.cfg` is invisible in editor.
*  After reloading the editor, everything back to normal.

### Steps to reproduce

- Create a new plugin, active by default. `plugin.cfg`  file likes below in my case.
```
[plugin]

name="My_plugin"
description=""
author=""
version=""
script="plugin.gd"
```
- Add the code below to `plugin.gd` which intend to support json file type at POT genaration 
```
@tool
extends EditorPlugin

class EditorTranslationParser extends EditorTranslationParserPlugin:
	
	func _parse_file(path, msgids, msgids_context_plural):
		var file = FileAccess.open(path, FileAccess.READ)
		var text = file.get_as_text()
		msgids.append(text)


	func _get_recognized_extensions():
		return ["json"]


var parser = EditorTranslationParser.new()


func _enter_tree() -> void:
	add_translation_parser_plugin(parser)


func _exit_tree() -> void:
	# Clean-up of the plugin goes here.
	remove_translation_parser_plugin(parser)

```
- The plugin is active by default, if you don't then active the plugin
- Create a dummy json file,  
- add this file to POT generation, 
- press Generate POT, then you will get error like ` editor/pot_generator.cpp:78 - Unrecognized file extension json in generate_pot()` even the plugin is actived
- Reload the editor
- Repeat the process of POT generate, everything works fine right now.

### Minimal reproduction project

N/A

**Comment by AnidemDex** [comment]
> The plugin is active by default
> press Generate POT, then you will get error like  `editor/pot_generator.cpp:78 - Unrecognized file extension json in generate_pot()` even the plugin is actived

That's... expected? Plugin doesn't have a "hot reload". If the plugin is active and you change the plugin script, the already active plugin **will not** call the virtual functions that you use to prepare the plugin (`_enter_tree` in example, because the generated node _is already_ in tree).

---

> - The new created plugin is not working with or without active.
> - plugin.cfg is invisible in editor.
> - After reloading the editor, everything back to normal.

This, in the other hand, is an issue (and doesn't seems related to reproduction project, or does it?). This didn't happened in Godot 4.0.x, I'll test with most recent release and see if that still happens

**Comment by 1Googol** [comment]
To be clear, English is not my native language, so my description may not be accurate and maybe misunderstand your reply.

> 
> That's... expected? Plugin doesn't have a "hot reload". If the plugin is active and you change the plugin script, the already active plugin **will not** call the virtual functions that you use to prepare the plugin (`_enter_tree` in example, because the generated node _is already_ in tree).
> 

I belive this is expected. 
- Check  docs [Making plugins](https://docs.godotengine.org/en/stable/tutorials/plugins/editor/making_plugins.html), you can find sentence like `
A plugin is a great way to extend the editor with useful tools. It can be made entirely with GDScript and standard scenes, without even reloading the editor.`. 
- And after things back to normal, I'm be able to change the plugin script with plugin actived and save changes and the changes work. (ie. change `return ["json"]` to `return ["json", "aaa"], save changes,  POT Generation will also add context from `*.aaa` files you added  to the  `*.pot` file it generated.)

> 
> This, in the other hand, is an issue (and doesn't seems related to reproduction steps, or does it?). This didn't happened in Godot 4.0.x, I'll test with most recent release and see if that still happens
>
You can just simply add ` print("enter tree")` to `_enter_tree()`  instead  of the code I offered. But with my specific case, you can get clearly erro info. And I also mention things like `plugin.cfg is invisible in editor.` I saw, because [Making plugins](https://docs.godotengine.org/en/stable/tutorials/plugins/editor/making_plugins.html)  shows visible `plugin.cfg` file in editor, maybe it matters maybe not, I just put it there in case

       
  
  
**To be cautious, I reconfirmed it and got some different phenomena than before**
* It still print erro `editor/pot_generator.cpp:78 - Unrecognized file extension json in generate_pot()` (New created plugin, actived).
* But except reloading editor, you can also toggle plugin active status twice to get it work. As I can recall,  last time I test it  with plugin active status toggle is useless. I don't know why, but  that's what happed.

**Comment by AnidemDex** [comment]
TL;DR Can't replicate

---

> But except reloading editor, you can also toggle plugin active status twice to get it work. 

Toggling the state of the plugin (reloading the plugin) is the easiest way to make the changes you made on _plugin script_ applied, that's my initial point on previous comment.

> Check docs [Making plugins](https://docs.godotengine.org/en/stable/tutorials/plugins/editor/making_plugins.html), you can find sentence like  A plugin is a great way to extend the editor with useful tools. It can be made entirely with GDScript and standard scenes, without even reloading the editor..

Yes, plugins extends editor functionality without needing to reload the editor, but in the same documentation is advised to _not_ have the plugin active the first time and then activating it _after_ adding the custom node code:
![image](https://github.com/godotengine/godot/assets/7025991/027e6b48-2d6c-4425-b65d-73c35ae90c97)

![image](https://github.com/godotengine/godot/assets/7025991/1fdc0bc4-bcb1-4ab3-a3b3-d76cd7b75ce6)


---

Just to be sure, I tried creating a new _empty_ plugin on v4.1.2.stable.official [399c9dc39] following the linked documentation page:
![image](https://github.com/godotengine/godot/assets/7025991/f9769b1c-2bb7-495e-80e9-518cce0d1ffb)

The folder, plugin script and script configuration (`plugin.cfg`) files are created:
![image](https://github.com/godotengine/godot/assets/7025991/9dcab8b5-20b2-4dc0-9f2c-b4ff248789ff)

I've added the code you provided to plugin script and activated the plugin, then I've created an empty json file. After that, just selected that empty json and generated the POT file with no errors:
![image](https://github.com/godotengine/godot/assets/7025991/ede24895-f400-4881-8bce-ebdd28d10d5a)

![image](https://github.com/godotengine/godot/assets/7025991/82671cc8-bc6a-4c1e-b14f-4f4d0f88a2d2)

**Comment by 1Googol** [comment]
> Toggling the state of the plugin (reloading the plugin) is the easiest way to make the changes you made on _plugin script_ applied, that's my initial point on previous comment.

So this is the problem. Yes, you are right, reloading the plugin works. But in my situation which I can't reproduce now, only reloading the editor be able to make it work, that is why this issue is here in the first place and the tittle before only mention reload editor to active plugin and probably why you put the first comment. I didn't record last time because I think it's easy to repreduce(I didn't know this is an unusual situation since this is my first plugin), well I guess I was wrong, so let me put a video below to show what I can still reproduce on `v4.1.3.stable.official [f06b6836a]`:
- Can reproduce: error debug info print when the plugin is first created with active  when create the plugin.
- Can reproduce: `plugin.cfg` is invisible when the plugin is first created.

https://github.com/godotengine/godot/assets/72542400/5b47e259-6bc5-47eb-8b5c-674e4636ddce


So after all, I think you are probably right, this is a problem of usage now, I will just close the comment.

**Comment by AnidemDex** [comment]
If you need help with plugin making, take a look at Godot Engine discord, on #plugin channel, I usually tend to be around there

**Comment by 1Googol** [comment]
That is Great.
